.github/workflows/cypress-tests-runner.yml

```diff
@@ -60,6 +60,41 @@ jobs:
         if: ${{ env.RUN_TESTS == 'true' }}
         uses: actions/checkout@v4
 
+      - name: Install Node.js
+        if: ${{ env.RUN_TESTS == 'true' }}
+        uses: actions/setup-node@v4
+        with:
+          node-version: '20'
+          cache: 'npm'
+
+      - name: Install Cypress and dependencies
+        if: ${{ env.RUN_TESTS == 'true' }}
+        run: |
+          npm ci --prefix ./cypress-tests
+
+      - name: Check formatting
+        if: ${{ env.RUN_TESTS == 'true' }}
+        run: |
+          npm run format:check --prefix cypress-tests && npm run lint --prefix cypress-tests
+
+          if [ $? -ne 0 ]; then
+            echo "::notice::Formatting or lint check failed"
+            git config --local user.name 'hyperswitch-bot[bot]'
+            git config --local user.email '148525504+hyperswitch-bot[bot]@users.noreply.github.com'
+
+            # Run prettier to fix formatting
+            npm run format --prefix cypress-tests
+
+            # Run eslint to auto-fix linting issues
+            # This will not fix logical issues, but will fix formatting issues
+            npm run lint --prefix cypress-tests -- --fix
+
+            # Add changes to git and commit
```

Comment:
Committing changes would require you to checkout the branch correctly, and not a merge commit of the PR branch and base branch. The checkout step would have to be updated to handle this.